src/dotnet-svcutil/lib/src/FrameworkFork/Microsoft.Xml/Utils/StringConcat.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.ComponentModel;
+
+namespace Microsoft.Xml.Xsl.Runtime {
+
+    /// <summary>
+    /// Efficiently concatenates strings when the number of string is not known beforehand, and
+    /// yet it is common for only one string to be concatenated.  StringBuilder is not good for
+    /// this purpose, since it *always* allocates objects, even if only one string is appended.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public struct StringConcat {
+        private string s1, s2, s3, s4;
```

Comment:
Get rid of this class. It's irrelevant as this is a dev time tool and not runtime. Caching a StringBuilder would be a better option.